app/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDao.java

```diff
@@ -7,68 +7,84 @@
 import com.amazonaws.services.dynamodbv2.model.AttributeValue;
 import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
 import com.amazonaws.services.dynamodbv2.model.Condition;
-import com.amazonaws.services.dynamodbv2.model.ConditionalCheckFailedException;
 import com.google.common.collect.Sets;
 
 @Component
 public class DynamoUserConsentDao implements UserConsentDao {
 
-    private DynamoDBMapper mapper;
+    private DynamoDBMapper mapperV2;
+    private DynamoDBMapper mapperV3;
+
+    @Resource(name = "userConsentDdbMapper2")
+    public final void setDdbMapper2(DynamoDBMapper mapper2) {
+        this.mapperV2 = mapper2;
+    }
 
-    @Autowired
-    public void setDynamoDbClient(BridgeConfig bridgeConfig, AmazonDynamoDB client) {
-        DynamoDBMapperConfig mapperConfig = new DynamoDBMapperConfig.Builder().withSaveBehavior(SaveBehavior.UPDATE)
-                .withConsistentReads(ConsistentReads.CONSISTENT)
-                .withTableNameOverride(DynamoUtils.getTableNameOverride(DynamoUserConsent2.class, bridgeConfig)).build();
-        mapper = new DynamoDBMapper(client, mapperConfig);
+    @Resource(name = "userConsentDdbMapper3")
+    public final void setDdbMapper3(DynamoDBMapper mapper3) {
+        this.mapperV3 = mapper3;
     }
 
     @Override
     public UserConsent giveConsent(String healthCode, StudyConsent studyConsent) {
         checkArgument(isNotBlank(healthCode), "Health code is blank or null");
         checkNotNull(studyConsent);
-        DynamoUserConsent2 consent = null;
-        try {
-            consent = getUserConsent(healthCode, studyConsent);
-            if (consent == null) {
-                consent = new DynamoUserConsent2(healthCode, studyConsent);
-            }
-            consent.setSignedOn(DateTime.now(DateTimeZone.UTC).getMillis());
-            mapper.save(consent);
-        } catch (ConditionalCheckFailedException e) {
-            throw new EntityAlreadyExistsException(consent);
+
+        // This will be passed in to the method when we coordinate Stormpath and DDB operations
+        long signedOn = DateUtils.getCurrentMillisFromEpoch();
+
+        // It doesn't currently matter which table your consent is in, you can't consent again 
+        // if a record exists.
+        UserConsent existingConsent = getUserConsent(healthCode, new StudyIdentifierImpl(studyConsent.getStudyKey()));
+        if (existingConsent != null) {
+            throw new EntityAlreadyExistsException(null, "Consent already exists.");
         }
+        
+        DynamoUserConsent3 consent = new DynamoUserConsent3(healthCode, studyConsent.getStudyKey());
+        consent.setConsentCreatedOn(studyConsent.getCreatedOn());
+        consent.setSignedOn(signedOn);
+        mapperV3.save(consent);
+
         return consent;
     }
 
     @Override
     public boolean withdrawConsent(String healthCode, StudyIdentifier studyIdentifier) {
```

Comment:
Do we want withdrawConsent() to just call deleteConsent()? Semantically, they do the same thing.